mephisto/operations/supervisor.py

```diff
@@ -211,7 +211,8 @@ def shutdown(self):
         if self.sending_thread is not None:
             self.sending_thread.join()
         logger.debug(f"Joining agents {self.agents}")
-        for agent_info in self.agents.values():
+        agents_to_close = list(self.agents.values())
```

Comment:
Could you add a comment explaining the reason for separating it like that here. If I am refactoring this code and I get to to this line, I definitely change it back to what it was before this PR.